generator/src/main/java/com/scottlogic/deg/generator/restrictions/NumericRestrictions.java

```diff
@@ -36,6 +36,23 @@ public boolean match(Object o) {
         return true;
     }
 
+    public boolean numericValuesAreInteger() {
+        if (min == null || max == null) {
+            return true;
+        }
+
+        // If either of the min or max values have decimal points or if the sign differs when converting to an integer
+        // the value is not an integer
+        if ((min.getLimit().scale() > 0 || max.getLimit().scale() > 0) ||
+            (min.getLimit().signum() != Integer.signum(min.getLimit().intValue()) ||
```

Comment:
can min.getLimit() and max.getLimit()  be local variables as they are called lots here. Also if all of these statements are OR'd together I don't think there is any need for the brackets.